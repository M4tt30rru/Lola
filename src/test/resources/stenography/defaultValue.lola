##Import "std.lola"
##Find
 ##MethodDeclaration(dec) = default;
 ##run {
__ = dec.t.n.name.strip()
if __ == 'void':
 ret = None
elif __ in ['int', 'short', 'byte']:
 ret = '0'
elif __ in ['double', 'float']:
 ret = '0.0'
elif __ in ['boolean']:
 ret = 'false'
else:
 ret = 'null'
 }
 ##replace ##(dec) {##('' if ret is None else 'return ' + ret + ';')}

double f() = default;
int g() = default;
T h() = default;
void k() = default;