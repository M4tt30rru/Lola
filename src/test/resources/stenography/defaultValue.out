double f() {
	return 0.0;
}

int g() {
	return 0;
}

T h() {
	return null;
}

void k() {
}